test/fw/ptl/utils/pbs_anonutils.py

```diff
@@ -734,19 +743,176 @@ def __delete_kv(self, line, key, value):
+        elif key in self.gmap_resc_key:
+            self.gmap_resc_key[alias_key] = self.gmap_resc_key[key]
+        elif key in self.gmap_resc_val:
+            self.gmap_resc_val[alias_key] = self.gmap_resc_val[key]
+
+        # Then check the input maps
+        if key in self.attr_delete:
+            self.attr_delete[alias_key] = self.attr_delete[key]
+        if key in self.attr_key:
+            self.attr_key[alias_key] = self.attr_key[key]
+        if key in self.attr_val:
+            self.attr_val[alias_key] = self.attr_val[key]
+        if key in self.resc_delete:
+            self.resc_delete[alias_key] = self.resc_delete[key]
+        if key in self.resc_key:
+            self.resc_key[alias_key] = self.resc_key[key]
+        if key in self.resc_val:
+            self.resc_val[alias_key] = self.resc_val[key]
+
+        return None
+
+    def anonymize_qstat_short(self, filename, extension=".anon",
+                              inplace=False):
+        """
+        Anonymize qstat short format outputs (tabular form)
+        The 'titles' of various columns are used to look up keys inside the
+        global attribute maps and they are anonymized/removed accordingly.
+
+        :param filename: Name of the file to anonymize
+        :type filename: str
+        :param delim: delimiter for the table
+        :type delim: str
+        :param extension: Extension of the anonymized file
+        :type extension: str
+        :param inplace: If true returns the original file name for
+                        which contents have been replaced
+        :type inplace: bool
 
-    def anonymize_file(self, filename, extension=".anon", inplace=False):
+        :returns: a str object containing filename of the anonymized file
         """
+        if not inplace:
+            fn = (filename + extension)
+            nf = open(fn, "w")
+        else:
+            (fd, fn) = self.du.mkstemp()
+            nf = os.fdopen(fd, "w")
+
+        # qstat outputs sometimes have different names for some attributes
+        self.__add_alias_attr(ATTR_euser, "User")
+        self.__add_alias_attr(ATTR_euser, "Username")
+        self.__add_alias_attr(ATTR_name, "Jobname")
+        self.__add_alias_attr(ATTR_name, "Name")
+
+        header = None
+        with open(filename) as f:
+            # Get the header and the line with '-'s
+            # Also write out the header and dash lines to the output file
+            line_num = 0
+            for line in f:
+                nf.write(line)
+                line_num += 1
+                line_strip = line.strip()
+
+                if line_strip[0].isalpha():
+                    header = line
+                    continue
+                # Dash line is the line after header
+                if header is not None:
+                    dash_line = line
+                    break
+
+            if header is None:  # Couldn't find the header
+                return filename
```

Comment:
Please open nf using a with statement as you did with f. Then you won't have to worry about closing it.